src/gem.sol

```diff
@@ -117,8 +117,8 @@ contract Gem {
     {
         unchecked {
             ok              = true;
-            balanceOf[dst] += wad;
             uint256 prevB   = balanceOf[src];
+            balanceOf[dst] += wad;
             balanceOf[src]  = prevB - wad;
```

Comment:
hm is this right?  looks like it might be introducing a bug

transferfrom(ali, ali, wad) => balanceOf(ali) == prevB - wad, so balance decreases
can you add a test for this?